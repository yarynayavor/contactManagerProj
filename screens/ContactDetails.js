import React from 'react';
import {Text, View, Image, StyleSheet, Button, TouchableHighlight, AsyncStorage, Alert} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import IconF from 'react-native-vector-icons/MaterialIcons';
import uuid from 'uuid';


export default class ContactDetails extends React.Component {

    constructor() {
        super();
        this.state = {
            contact: [],
            contacts: [],
            favorites:[]
        }

        this.onPressFavorites = this.onPressFavorites.bind(this);
        this.deleteContact = this.deleteContact.bind(this);
    }

    componentDidMount() {
        let {contacts} = this.state;
        this.setState({ contact: this.props.navigation.state.params.contact });
        AsyncStorage.getItem('favorites').then((data) => {
            if(data) {
                this.setState({
                    favorites: JSON.parse(data),
                });
            }
        });

        AsyncStorage.getItem('contacts').then((data) => {
            if(data) {
                this.setState({
                    contacts: JSON.parse(data),
                });
            } else {
                this.setState({
                    contacts
                });
            }
        });
    }

    deleteContact(_id) {
        let {contacts,favorites} = this.state;
        let _contacts = contacts.filter(c => c.id != _id);
        let _favorites = favorites.filter(c => c.id != _id);
        Alert.alert(
            'Contact manager',
            'Contact deleted',
            [
                {text: 'OK', onPress: () => console.log('OK Pressed')},
            ],
            { cancelable: false }
        );
        AsyncStorage.setItem('contacts', JSON.stringify(_contacts));
        AsyncStorage.setItem('favorites', JSON.stringify(_favorites));
        this.setState({
            contacts: _contacts,
            favorites: _favorites

        })
        this.props.navigation.navigate("ContactList", {contacts: contacts});
    }

    onPress() {
        this.props.navigation.navigate("ContactList");
    }

    checkDublicates(contact) {
        let {favorites} = this.state;
        for (var i = 0; i < favorites.length; i++) {
            if(favorites[i].cellPhone===contact.cellPhone) {
                return true;
            }
        }
        return false;
    }

    onPressFavorites(contact) {
        let {favorites} = this.state;

        let checkdubl=this.checkDublicates(contact);
        if(checkdubl) {
            Alert.alert(
                'Contact manager',
                'Contact already exist in favorites!',
                [
                    {text: 'OK', onPress: () => console.log('OK Pressed')},
                ],
                { cancelable: false }
            );
        } else {
            Alert.alert(
                'Contact manager',
                'Contact added to favorites!',
                [
                    {text: 'OK', onPress: () => console.log('OK Pressed')},
                ],
                { cancelable: false }
            );
            favorites.unshift({
                firstName: contact.firstName,
                lastName: contact.lastName,
                cellPhone: contact.cellPhone,
                id:contact.id
            });
        }

        this.setState({favorites });
        AsyncStorage.setItem('favorites', JSON.stringify(favorites));
    }

    render() {
        return (
            <View style={styles.container}>
                {this.state.contact !== 0 ?
                    <View>
                        <View>
                            <View style={styles.skyBlue}>
                                <TouchableHighlight underlayColor='#7fe2ff' onPress={()=>{this.onPress()}} style={{width:50}}>
                                    <View>
                                        <Icon style={styles.addButton} name="arrow-left" size={20} color="#fff"/>
                                    </View>
                                </TouchableHighlight>
                            </View>
                            <View style={styles.imageWrapper}>
                                <Icon style={styles.image} name="user-circle" size={100} color="#b2edff"/>

                            </View>
                            <Text style={styles.name}>{this.state.contact.firstName} {this.state.contact.lastName}</Text>
                        </View>
                        <View>
                            <View style={styles.phoneWrapper}>
                                <Text style={styles.metadata}>Phone: </Text>
                                <Text style={styles.detailData}>{this.state.contact.cellPhone}</Text>
                            </View>
                        </View>
                        <TouchableHighlight onPress={()=>{this.onPressFavorites(this.state.contact)}}>
                            <View>
                                <Text style={styles.addToFavorites}>
                                    <IconF name="favorite" size={22} color="red"/>
                                    Add to favorites</Text>
                            </View>
                        </TouchableHighlight>
                        <TouchableHighlight style={{marginTop:5}} onPress={()=>{this.deleteContact(this.state.contact.id)}}>
                            <View>
                                <Text style={styles.deleteContact}>
                                    <Icon name="trash" size={22} color="#fff"/>
                                     Remove contact</Text>
                            </View>
                        </TouchableHighlight>
                    </View>
                    : null}
            </View>)
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor:'#e5f9ff'
    },
    skyBlue: {
        backgroundColor: 'skyblue',
        height: 40
    },
    imageWrapper: {
        alignItems: 'center',
        marginBottom: 5
    },
    image: {
        width: 100,
        height: 100,
        borderWidth: 1,
        borderColor: '#004e66',
        borderRadius: 100,
        marginTop:5
    },
    name: {
        fontSize: 35,
        fontWeight: '800',
        color: '#004e66',
        textAlign:'center'
    },
    phoneWrapper: {
        flexDirection: 'row',
        backgroundColor: '#f2fcff',
        padding:10,
        paddingTop:15,
        paddingBottom:15,
        marginBottom:30
    },
    metadata: {
        fontWeight: '600',
        fontSize: 15,
        width: 120,
        marginRight: 5,
        fontStyle:'italic'
    },
    detailData: {
        fontSize: 25,
        textAlign:'center',
    },
    addButton: {
        textAlign:'center',
        backgroundColor:'#ccf3ff',
        padding:10,
        color:'#00c5ff',
    },
    addToFavorites: {
        textAlign:'center',
        backgroundColor:'skyblue',
        padding:5,
        color:'#fff',
        fontSize:22,
        borderWidth: 1,
        borderColor: '#004e66',
        fontStyle:'italic'
    },
    deleteContact: {
        textAlign:'center',
        backgroundColor:'#ff4c4c',
        color:'#fff',
        padding:5,
        fontSize:22,
        borderWidth: 1,
        borderColor: '#004e66'
    },
});