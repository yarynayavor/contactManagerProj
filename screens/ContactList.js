import React from 'react';
import { Text, View,
    StyleSheet,
    ScrollView,
    TouchableNativeFeedback,AsyncStorage,TextInput,
    TouchableHighlight,
    Alert} from 'react-native';
import uuid from 'uuid';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class ContactList extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            contacts: [],
            text: '',
            logs:[],
            searchInput:'',
            filteredArray:[]
        }

        this.onPressDetails = this.onPressDetails.bind(this);
        this.onPressCall = this.onPressCall.bind(this);
        this.onPressAddContact=this.onPressAddContact.bind(this);
    }

    capitalize(val) {
        return val[0].toUpperCase()+val.substring(1);
    }

    componentDidMount() {
        let {contacts} = this.state;
        //AsyncStorage.removeItem('contacts');
        AsyncStorage.getItem('contacts').then((data) => {
            if(data) {
                this.setState({
                    contacts: JSON.parse(data),
                });
            } else {
                fetch("https://randomuser.me/api/?nat=gb&inc=name,email,cell,picture&results=20").then(x => {
                    const results = JSON.parse(x._bodyInit).results;
                    results.map((contact) => {
                        contacts.push({
                            firstName: this.capitalize(contact.name.first),
                            lastName: this.capitalize(contact.name.last),
                            cellPhone: contact.cell,
                            id: uuid()
                        });
                        this.setState({ text:contact.cell});
                        return contacts;
                    })
                    this.setState({ contacts});
                    AsyncStorage.setItem('contacts', JSON.stringify(contacts));
                });
            }

        });

        AsyncStorage.getItem('logs').then((data) => {
            if(data) {
                this.setState({
                    logs: JSON.parse(data),
                });
            }
        });
    }
    onPressDetails(contact) {
        this.props.navigation.navigate("ContactDetails", {contact: contact});
    }

    onPressAddContact(contact) {
        this.props.navigation.navigate("AddNewContact", {contact: contact});
    }

    onPressCall(contact) {
        let {logs} = this.state;
        Alert.alert(
            'Contact manager',
            'You make a call, see in Logs',
            [
                {text: 'OK', onPress: () => console.log('OK Pressed')},
            ],
            { cancelable: false }
        );

        logs.unshift({
            firstName: contact.firstName,
            lastName: contact.lastName,
            cellPhone: contact.cellPhone,
            callDuration:'00:00:01'
        });
        this.setState({logs });
        AsyncStorage.setItem('logs', JSON.stringify(logs));
    }

    drawContent(contact, index) {
        return (
            <TouchableNativeFeedback key={index} onPress={()=> {this.onPressDetails(contact)}}>
                <View style={styles.contact}>
                    <Icon style={styles.image} name="user-circle" size={35} color="#b2edff"/>
                    <View style={styles.contactUser}>
                        <Text style={styles.contactName}>
                            <Text>{contact.firstName} </Text>
                            <Text>{contact.lastName}</Text>
                        </Text>
                        <Text style={styles.contactCell}>{contact.cellPhone}</Text>
                    </View>
                    <TouchableHighlight underlayColor='#99ff9a' style={styles.dial} onPress={()=>{this.onPressCall(contact)}}>
                        <Icon  name="phone" size={35} color="#00cc03"/>
                    </TouchableHighlight>
                </View>
            </TouchableNativeFeedback>
        );
    }
    render() {
        const {contacts,searchInput}=this.state;
        let filteredArr = contacts.filter(contact =>{
            let phoneString=contact.cellPhone.toString();
            let outString = phoneString.replace(/[-]/gi, '');
            return outString.includes(searchInput);
        })

        let res;
        if(filteredArr.length===0 && this.state.searchInput){
            res=<Text style={styles.text}>Nothing found</Text>;
        } else if (filteredArr.length===0) {
            res=<ScrollView style={styles.wrapper}>
                {contacts && contacts.map((contact, index) => {
                    return this.drawContent(contact, index)
                })}
            </ScrollView>;
        } else {
            res=<ScrollView style={styles.wrapper}>
                {filteredArr && filteredArr.map((filt, index) => {
                    return this.drawContent(filt, index)
                })}
            </ScrollView>;
        }

        return(
            <View style={styles.container}>
                <View style={styles.searchSection}>
                    <Icon style={styles.searchIcon} name="search" size={20} color="#ccc"/>
                    <TextInput
                        style={styles.input}
                        keyboardType={'phone-pad'}
                        placeholder="Search"
                        onChangeText={(searchInput) => this.setState({searchInput})}
                        underlineColorAndroid="transparent"
                        value={this.state.searchInput}
                    />
                </View>
                {res}
                <TouchableHighlight onPress={()=>{this.onPressAddContact()}}>
                    <View>
                        <Text style={styles.addButton}><Icon name="user-plus" size={35} color="#fff"/>
                            Add new contact</Text>
                    </View>
                </TouchableHighlight>
            </View>);
       }
}

const styles = StyleSheet.create({
    text: {
        fontSize:20,
        marginTop:10,
        textAlign:'center',
    },
    searchSection: {
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: '#eaeaea',
        padding:5,
    },
    searchIcon: {
        padding: 10,
    },
    input: {
        flex: 1,
        paddingLeft: 5,
        backgroundColor: '#fff',
        color: '#424242',
        fontSize:20
    },
    container: {
        flex: 1,
        backgroundColor:'#fff'
    },
    wrapper: {
        flex: 1,
        marginBottom: 10,
        marginTop:10,
    },
    contact: {
        flexDirection: 'row',
        borderBottomWidth: 1,
        borderBottomColor: '#849999',
        padding:10
    },

    contactUser: {
        paddingLeft:20
    },
    contactName: {
        fontWeight:'700',
        fontSize:18
    },
    contactCell: {
        fontSize:15
    },
    image: {
        marginLeft: 10,
        marginTop:10
    },
    dial: {
        marginLeft: 'auto',
        backgroundColor:'#e5f9ff',
        paddingLeft:15,
        paddingRight:15,
        paddingTop:5,
        paddingBottom:5,
        borderRadius:100
    },
    addButton: {
        textAlign:'center',
        backgroundColor:'#004e66',
        padding:5,
        color:'#fff'
    },
    fireSearch : {
        color:'red'
    },
});