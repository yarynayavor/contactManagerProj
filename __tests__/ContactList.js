import 'react-native';
import React from 'react';
import { shallow,mount } from 'enzyme';
import ContactList from '../screens/ContactList';
import jest from 'jest';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
import ContactDetails from "../screens/ContactDetails";

describe('Testing app', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <ContactList />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

