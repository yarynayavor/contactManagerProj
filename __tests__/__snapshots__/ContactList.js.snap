// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ContactList component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eaeaea",
        "flexDirection": "row",
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#ccc",
            "fontSize": 20,
          },
          Object {
            "padding": 10,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="Search"
      style={
        Object {
          "backgroundColor": "#fff",
          "color": "#424242",
          "flex": 1,
          "fontSize": 20,
          "paddingLeft": 5,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#004e66",
            "color": "#fff",
            "padding": 5,
            "textAlign": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        Add new contact
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing app renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eaeaea",
        "flexDirection": "row",
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#ccc",
            "fontSize": 20,
          },
          Object {
            "padding": 10,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="Search"
      style={
        Object {
          "backgroundColor": "#fff",
          "color": "#424242",
          "flex": 1,
          "fontSize": 20,
          "paddingLeft": 5,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#004e66",
            "color": "#fff",
            "padding": 5,
            "textAlign": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        Add new contact
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eaeaea",
        "flexDirection": "row",
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#ccc",
            "fontSize": 20,
          },
          Object {
            "padding": 10,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="Search"
      style={
        Object {
          "backgroundColor": "#fff",
          "color": "#424242",
          "flex": 1,
          "fontSize": 20,
          "paddingLeft": 5,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#004e66",
            "color": "#fff",
            "padding": 5,
            "textAlign": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        Add new contact
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eaeaea",
        "flexDirection": "row",
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#ccc",
            "fontSize": 20,
          },
          Object {
            "padding": 10,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="Search"
      style={
        Object {
          "backgroundColor": "#fff",
          "color": "#424242",
          "flex": 1,
          "fontSize": 20,
          "paddingLeft": 5,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#004e66",
            "color": "#fff",
            "padding": 5,
            "textAlign": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        Add new contact
      </Text>
    </View>
  </View>
</View>
`;
